util/log: cleanup TestGC

BramGruneir opened the pull request:
This test was failing on windows with a high regularity due to not flushing the logs before gcing them.
To make this test predictable, a mechanism to disable the gc was added.

BramGruneir committed:
util/log: cleanup TestGC

This test was failing on windows with a high regularity due to not flushing the logs before gcing them.
To make this test predictable, a mechanism to disable the gc was added.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/14831)
<!-- Reviewable:end -->

knz commented:
LGTM - but could the GC also be responsible for  #12750 (and #14147 #14014)? Could you have a brief look and think whether this is related?

---

Reviewed 3 of 3 files at r1.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-:-KhWHlOyYopZ2QJOLZ-r:b3qqnrq)*
<!-- Sent from Reviewable.io -->

tamird commented:
lots of magic here and vague comments about races that aren't sufficiently investigated.

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[pkg/util/log/clog_test.go, line 509 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUriTm-6bCepDI0Gh:-KhXUriTm-6bCepDI0Gi:b3kre) ([raw file](https://github.com/cockroachdb/cockroach/blob/e9a94ab76483813130d2c2c617cc921ecc6a434f/pkg/util/log/clog_test.go#L509)):*
> ```Go
> 
> 	setFlags()
> 	logging.stderrThreshold.set(Severity_NONE)
> ```

?

---

*[pkg/util/log/clog_test.go, line 513 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUG0qk_aloHlFP-fI:-KhXUG0qk_aloHlFP-fJ:b-a5xyp) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L513)):*
> // Start the first log file. If this is done afterwards, there's a chance of
> 	// of an extra log file being created during the subsequent logging.

Can you explain how that happens?

---

*[pkg/util/log/clog_test.go, line 541 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUUUqON10PtkUH-FL:-KhXUUUqON10PtkUH-FM:b-yevqf9) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L541)):*
> ```Go
> 	}
> 
> 	// Empirically, each log file is ~650 bytes in size. GC should trim this
> ```

there are a lot of magic numbers here. this comment should be up near the `atomic.StoreUint64(&MaxSizePerSeverity, 1500)`, and the constant `expectedFilesAfterGC` should be defined in terms of 1500 and this 650.

---

*[pkg/util/log/clog_test.go, line 547 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUw-OuVPOeqTPAod_:-KhXUw-OuVPOeqTPAoda:b-tj7b5q) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L547)):*
> ```Go
> 	}
> 
> 	// Flush all the logs or there is a slight race condition on windows in
> ```

again, what? what race condition? what is racing?

---

*[pkg/util/log/file.go, line 52 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXU7ym33eA9-gvtnkG:-KhXU7ym33eA9-gvtnkH:b-wot7ro) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/file.go#L52)):*
> ```Go
> 
> // enableGCDeamon can be used to turn off the logging GC.
> var enableGCDeamon int32 = 1
> ```

why isn't this in clog.go, next to where it is used?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-:-KhXV0EzWn2B6QwSWUmR:b1dyuhz)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
The magic was here before, but has been fixed now, and I omitted details of the races since it's referencing other functions and I was a little worried about code rot.

---

Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions.

---

*[pkg/util/log/clog_test.go, line 509 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUriTm-6bCepDI0Gh:-KhYaAEU3QgiWClZ1215:br2x9ys) ([raw file](https://github.com/cockroachdb/cockroach/blob/e9a94ab76483813130d2c2c617cc921ecc6a434f/pkg/util/log/clog_test.go#L509)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

?
</blockquote></details>

There was no need for this to be set.  So I removed it.

---

*[pkg/util/log/clog_test.go, line 513 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUG0qk_aloHlFP-fI:-KhYaKB0Cu5_WoGA6DiB:b-7uel0i) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L513)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

> // Start the first log file. If this is done afterwards, there's a chance of
> 	// of an extra log file being created during the subsequent logging.

Can you explain how that happens?
</blockquote></details>

Of course.  I've added more details to the comment.

---

*[pkg/util/log/clog_test.go, line 541 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUUUqON10PtkUH-FL:-KhYeWO-b3M0-LvVKASI:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L541)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

there are a lot of magic numbers here. this comment should be up near the `atomic.StoreUint64(&MaxSizePerSeverity, 1500)`, and the constant `expectedFilesAfterGC` should be defined in terms of 1500 and this 650.
</blockquote></details>

Done.

---

*[pkg/util/log/clog_test.go, line 547 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUw-OuVPOeqTPAod_:-KhYgnvpdPVIOp_pXQuT:bdf89m5) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L547)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

again, what? what race condition? what is racing?
</blockquote></details>

I've added details in the comment.

---

*[pkg/util/log/file.go, line 52 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXU7ym33eA9-gvtnkG:-KhYbKyqXFkSw0a1Ii60:bkncsjn) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/file.go#L52)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

why isn't this in clog.go, next to where it is used?
</blockquote></details>

Even better, I moved it under logging.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-:-KhYgrrzYn90144C9xQJ:b-kcuje3)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[pkg/util/log/clog_test.go, line 547 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUw-OuVPOeqTPAod_:-KhYiLLPh9MA-bRgD6nQ:b-bw5nqf) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L547)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

I've added details in the comment.
</blockquote></details>

why is this race specific to windows?

---

*[pkg/util/log/clog_test.go, line 521 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhYi2Qa77oRCXQ9A2NL:-KhYi2Qa77oRCXQ9A2NM:b-1y6ecw) ([raw file](https://github.com/cockroachdb/cockroach/blob/a79ac57cebb373da8dcd1d60da5d15876d106179/pkg/util/log/clog_test.go#L521)):*
> ```Go
> 
> 	// Start the first log file. If this is done afterwards, there's a chance of
> 	// of an extra log file being created during the subsequent logging based
> ```

im still utterly confused; why don't you disable the gc daemon first to nullify races, and then do whatever else you need to do?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/util/log/clog_test.go, line 547 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUw-OuVPOeqTPAod_:-KhYpknaOJ2iwiG2Dw0N:b-ow16ha) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L547)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

why is this race specific to windows?
</blockquote></details>

That I couldn't tell you.  I'm guessing difference in how the lower level libraries are implemented.

---

*[pkg/util/log/clog_test.go, line 521 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhYi2Qa77oRCXQ9A2NL:-KhYppUQfgvVgGmw3Tpf:bx0123u) ([raw file](https://github.com/cockroachdb/cockroach/blob/a79ac57cebb373da8dcd1d60da5d15876d106179/pkg/util/log/clog_test.go#L521)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

im still utterly confused; why don't you disable the gc daemon first to nullify races, and then do whatever else you need to do?
</blockquote></details>

This doesn't have to do with the gcdeamon.  In this case, it's the flushDeamon.  I could add a similar block on it if you prefer, but it seemed like overkill since this info write and the lockAndFlushAll syncs everything appropriately.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

tamird commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/util/log/clog_test.go, line 547 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUw-OuVPOeqTPAod_:-KhYrN0ChknorknSjgYH:bpff4yo) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L547)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

That I couldn't tell you.  I'm guessing difference in how the lower level libraries are implemented.
</blockquote></details>

I think we should probably investigate that, especially if there's a Go bug hiding under here. I can take this over, if you prefer.

---

*[pkg/util/log/clog_test.go, line 521 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhYi2Qa77oRCXQ9A2NL:-KhYrCkcr0DPoZLUJ_tc:bhbjl4z) ([raw file](https://github.com/cockroachdb/cockroach/blob/a79ac57cebb373da8dcd1d60da5d15876d106179/pkg/util/log/clog_test.go#L521)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

This doesn't have to do with the gcdeamon.  In this case, it's the flushDeamon.  I could add a similar block on it if you prefer, but it seemed like overkill since this info write and the lockAndFlushAll syncs everything appropriately.
</blockquote></details>

OK, so it's the flushDaemon. Why is this hack better than something explicit and self-documenting?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/util/log/clog_test.go, line 521 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhYi2Qa77oRCXQ9A2NL:-KhYrTxcd78KcVdIY_Th:bva1kgf) ([raw file](https://github.com/cockroachdb/cockroach/blob/a79ac57cebb373da8dcd1d60da5d15876d106179/pkg/util/log/clog_test.go#L521)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

OK, so it's the flushDaemon. Why is this hack better than something explicit and self-documenting?
</blockquote></details>

Agreed that this approach feels a bit fragile. I'd prefer to disable both of the background daemons for the test and make explicit calls to GC and flushing when necessary.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/util/log/clog_test.go, line 547 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUw-OuVPOeqTPAod_:-KhYsjmmdsslVyuHBZgB:b-z2gtjz) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L547)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

I think we should probably investigate that, especially if there's a Go bug hiding under here. I can take this over, if you prefer.
</blockquote></details>

I'd love to learn the best practices on how to do this instead of just handing it off.   Would you be up for pairing some time tomorrow or monday ?

---

*[pkg/util/log/clog_test.go, line 521 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhYi2Qa77oRCXQ9A2NL:-KhZ-bgLf4lku8fSEXNJ:b-1eh3wi) ([raw file](https://github.com/cockroachdb/cockroach/blob/a79ac57cebb373da8dcd1d60da5d15876d106179/pkg/util/log/clog_test.go#L521)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Agreed that this approach feels a bit fragile. I'd prefer to disable both of the background daemons for the test and make explicit calls to GC and flushing when necessary.
</blockquote></details>

Sure, done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

*[pkg/util/log/clog.go, line 1026 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ0yfQAcsE40GaNSgQ:-KhZ0yfQAcsE40GaNSgR:b-1vmkwr) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog.go#L1026)):*
> ```Go
> 	for range time.Tick(flushInterval) {
> 		if atomic.LoadInt32(&l.disableDeamons) == 0 {
> 			l.lockAndFlushAll()
> ```

so we already have a lock here, can we put `disableDaemons` behind that lock? pretty pretty please?

---

*[pkg/util/log/clog.go, line 1051 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ13t2fuAvUZ9ylCqB:-KhZ13t2fuAvUZ9ylCqC:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog.go#L1051)):*
> ```Go
> 	for range l.gcNotify {
> 		if atomic.LoadInt32(&l.disableDeamons) == 0 {
> 			l.gcOldFiles()
> ```

ditto

---

*[pkg/util/log/clog_test.go, line 547 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhXUw-OuVPOeqTPAod_:-KhZ1GN9gn3ZVwuMWD36:b-obzcqs) ([raw file](https://github.com/cockroachdb/cockroach/blob/cca0b83318c4678ea9bfa2ac57950ae4d5c247f9/pkg/util/log/clog_test.go#L547)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

I'd love to learn the best practices on how to do this instead of just handing it off.   Would you be up for pairing some time tomorrow or monday ?
</blockquote></details>

Yep, of course. I think there's no much magic here, though; just print statement debugging.

Please feel free to put something on my calendar.

---

*[pkg/util/log/clog_test.go, line 510 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ16QOAz4CnhDBj836:-KhZ16QOAz4CnhDBj837:bv7har9) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog_test.go#L510)):*
> ```Go
> 		atomic.StoreInt32(&logging.disableDeamons, previous)
> 	}(logging.disableDeamons)
> 	atomic.StoreInt32(&logging.disableDeamons, 1) // Turn off the gc.
> ```

remove this comment which has already rotted.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions.

---

*[pkg/util/log/clog.go, line 1026 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ0yfQAcsE40GaNSgQ:-Kh_Tv8i5ZQ-0Fkz19Iu:bftryk) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog.go#L1026)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

so we already have a lock here, can we put `disableDaemons` behind that lock? pretty pretty please?
</blockquote></details>

Done, but I'm concerned that now we'll lock when gcing files.  This may adversely affect logging performance.

---

*[pkg/util/log/clog.go, line 1051 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ13t2fuAvUZ9ylCqB:-Kh_UkcO8o2t2hcrwISa:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog.go#L1051)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---

*[pkg/util/log/clog_test.go, line 510 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ16QOAz4CnhDBj836:-Kh_Tsdo0pH8GLPrkPkQ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog_test.go#L510)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

remove this comment which has already rotted.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Also tested on windows and this still fixes the test.

---

Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-:-Kh_ZP7uy2EOlgteYhXF:b-lzacf2)*
<!-- Sent from Reviewable.io -->

knz commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[pkg/util/log/clog.go, line 1051 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ13t2fuAvUZ9ylCqB:-KhamB1sAJJ0ImwfPTS4:b-hhxunj) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog.go#L1051)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Can you put the conditional inside `gcOldFiles()`, i.e. `if l.disableDaemons { return }`

---

*[pkg/util/log/clog.go, line 631 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhamI5hKqlCxntxxb12:-KhamI5ip5U-AyO_kMgi:bkkk6zj) ([raw file](https://github.com/cockroachdb/cockroach/blob/54d22ed7447fd26e627dce5e0c7e3165b6b77131/pkg/util/log/clog.go#L631)):*
> ```Go
> 	traceLocation traceLocation
> 	// disableDeamons can be used to turn off both the GC and flush deamons.
> 	disableDeamons bool
> ```

I think we spell it "Daemons" usually.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[pkg/util/log/clog.go, line 1051 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ13t2fuAvUZ9ylCqB:-KhbbLgeEaX1prkFOWNG:bej1c5o) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog.go#L1051)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Can you put the conditional inside `gcOldFiles()`, i.e. `if l.disableDaemons { return }`
</blockquote></details>

Don't we call `gcOldFiles` in tests directly? I think this is fine as-is.

---

*[pkg/util/log/clog.go, line 1030 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhbbGqAOKnHVqBhJqUv:-KhbbGqBaJTc5bEBGYHz:b5l3mfi) ([raw file](https://github.com/cockroachdb/cockroach/blob/54d22ed7447fd26e627dce5e0c7e3165b6b77131/pkg/util/log/clog.go#L1030)):*
> ```Go
> func (l *loggingT) lockAndFlushAll() {
> 	l.mu.Lock()
> 	defer l.mu.Unlock()
> ```

why change this?

---

*[pkg/util/log/clog_test.go, line 511 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhbbVdseosM2VUFFab_:-KhbbVdtS30H2oZS0HM7:bv4q374) ([raw file](https://github.com/cockroachdb/cockroach/blob/54d22ed7447fd26e627dce5e0c7e3165b6b77131/pkg/util/log/clog_test.go#L511)):*
> ```Go
> 	defer func(previous bool) {
> 		logging.mu.Lock()
> 		defer logging.mu.Unlock()
> ```

nit: why defer? random inconsistency, i think

---

*[pkg/util/log/clog_test.go, line 547 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-Khbbi32jFxTIKqNylM2:-Khbbi32jFxTIKqNylM3:b-gy9q2k) ([raw file](https://github.com/cockroachdb/cockroach/blob/54d22ed7447fd26e627dce5e0c7e3165b6b77131/pkg/util/log/clog_test.go#L547)):*
> ```Go
> 		Infof(context.Background(), "%d", i)
> 		logging.mu.Lock()
> 		logging.flushAll()
> ```

what's the difference between this and Flush()? looks identical to me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-:-KhbbxSvKHWAsa6uBiob:bnfp4nl)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 0 of 2 files reviewed at latest revision, 6 unresolved discussions.

---

*[pkg/util/log/clog.go, line 1051 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhZ13t2fuAvUZ9ylCqB:-KhbdQfArt-tRPpFlUSa:bks7dlf) ([raw file](https://github.com/cockroachdb/cockroach/blob/e81a53419bc694113848890de6bf4261ac873d0f/pkg/util/log/clog.go#L1051)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

Don't we call `gcOldFiles` in tests directly? I think this is fine as-is.
</blockquote></details>

We do.  And if it was in gcOldFiles it wouldn't be disabling the daemon, but the function.  (I just fixed this for the flush daemon.)

---

*[pkg/util/log/clog.go, line 631 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhamI5hKqlCxntxxb12:-KhbcmlfSdJ8U1j-8ML8:b-ac7m6f) ([raw file](https://github.com/cockroachdb/cockroach/blob/54d22ed7447fd26e627dce5e0c7e3165b6b77131/pkg/util/log/clog.go#L631)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I think we spell it "Daemons" usually.
</blockquote></details>

Oh!  Fixed.  I'm surprised my spellchecker didn't catch this.

---

*[pkg/util/log/clog.go, line 1030 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhbbGqAOKnHVqBhJqUv:-Khbd8_S_XfXuUK-ILMy:bmsagd5) ([raw file](https://github.com/cockroachdb/cockroach/blob/54d22ed7447fd26e627dce5e0c7e3165b6b77131/pkg/util/log/clog.go#L1030)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

why change this?
</blockquote></details>

I thought this was our preferred style.  Makes no difference here.

---

*[pkg/util/log/clog_test.go, line 511 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-KhbbVdseosM2VUFFab_:-KhbegGdOeVHgPbMlVz-:bzawsln) ([raw file](https://github.com/cockroachdb/cockroach/blob/54d22ed7447fd26e627dce5e0c7e3165b6b77131/pkg/util/log/clog_test.go#L511)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

nit: why defer? random inconsistency, i think
</blockquote></details>

As I mentioned before, I thought defers were preferred.  Anyway, changed it.

---

*[pkg/util/log/clog_test.go, line 547 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-Khbbi32jFxTIKqNylM2:-KhbeCANLIjDrlRH1Xvy:bfgkk4k) ([raw file](https://github.com/cockroachdb/cockroach/blob/54d22ed7447fd26e627dce5e0c7e3165b6b77131/pkg/util/log/clog_test.go#L547)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

what's the difference between this and Flush()? looks identical to me.
</blockquote></details>

Flush was calling lockAndFlushAll, which was were the disabling was (incorrectly).  I've moved the disabling to the daemon itself, so it's safe to call flush again.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->

knz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831#-:-Khbj8IUGOIQyZZcxCn6:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14831)*
<!-- Sent from Reviewable.io -->
